plasmapy/physics/distribution.py

```diff
@@ -70,25 +77,286 @@ def Maxwellian_1D(v: units.m/units.s,
+    vThSq = vTh ** 2
+    # square of particle velocity [m^2/s^2]
+    vSq = (v - v_drift) ** 2
+    # calculating distribution function
+    try:
+        coeff1 = (np.pi * vThSq) ** (-3 / 2)
+        coeff2 = 4 * np.pi * vSq
+        expTerm = np.exp(-vSq / vThSq)
+        distFunc = coeff1 * coeff2 * expTerm
+    except Exception:
+        raise ValueError("Unable to get distribution function.")
+    return distFunc.to(units.s/units.m)
+
+@units.quantity_input
+def Maxwellian_speed_3D(vx: units.m/units.s,
+                        vy: units.m/units.s,
+                        vz: units.m/units.s,
+                        T, 
+                        particle="e",
+                        vx_drift=0*units.m/units.s,
+                        vy_drift=0*units.m/units.s,
+                        vz_drift=0*units.m/units.s):
+    r"""Return the probability of finding a particle with speed components
+    `v_x`, `v_y`, and `v_z`in m/s in an equilibrium plasma of temperature 
+    `T` which follows the 3D Maxwellian distribution function. This 
+    function assumes Cartesian coordinates.
+
+    Parameters
+    ----------
+    vx: Quantity
+        The speed in x-direction units convertible to m/s
+        
+    vy: Quantity
+        The speed in y-direction units convertible to m/s
+        
+    vz: Quantity
+        The speed in z-direction units convertible to m/s
+
+    T: Quantity
+        The temperature, preferably in Kelvin
+
+    particle: string, optional
+        Representation of the particle species(e.g., 'p' for protons, 'D+'
+        for deuterium, or 'He-4 +1' for $He_4^{+1}$ : singly ionized helium-4),
+        which defaults to electrons.
+        
+    vx_drift: Quantity
+        The drift speed in x-direction units convertible to m/s
+        
+    vy_drift: Quantity
+        The drift speed in y-direction units convertible to m/s
+        
+    vz_drift: Quantity
+        The drift speed in z-direction units convertible to m/s
+        
+    Returns
+    -------
+    f : Quantity
+        probability in speed^-1, normalized so that:
+        $\iiint_{0}^{\infty} f(\vec{v}) d\vec{v} = 1}
+
+    Raises
+    ------
+    TypeError
+        The parameter arguments are not Quantities and
+        cannot be converted into Quantities.
+
+    UnitConversionError
+        If the parameters is not in appropriate units.
+
+    ValueError
+        If the temperature is negative, or the particle mass or charge state
+        cannot be found.
+
+    Notes
+    -----
+    In one dimension, the Maxwellian speed distribution function describing
+    the distribution of particles with speed v in a plasma with temperature T
+    is given by:
+
+    .. math::
+    f = 4 \pi \vec{v}^2 (\pi * v_Th^2)^{-3/2} \exp(-(\vec{v} - \vec{v_{drift}})^2 / v_Th^2)
+    where v_Th = \sqrt(2 k_B T / m) is the thermal speed
+
+    See also
+    --------
+    Maxwellian_speed_1D
+
+    Example
+    -------
+    
```

Comment:
As previously, a doctest would be nice! We could go with a single particle for simplicity.